aten/src/ATen/NamedTensorUtils.cpp

```diff
@@ -60,6 +61,47 @@ void internal_set_names_inplace(Tensor& tensor, optional<DimnameList> names) {
   }
 }
 
+// Returns "Tensor['N', 'C', 'H', 'W']" for a tensor with names ('N', 'C', 'H', 'W').
```

Comment:
At some point it might be nice to have a dictionary-style output, e.g. Tensor[ N:5, C:10, ....]